wfc_dylib/src/lib.rs

```diff
@@ -442,25 +451,50 @@ pub enum WfcObserveResult {
 pub extern "C" fn wfc_observe(
     wfc_world_state_handle: WfcWorldStateHandle,
     wfc_rng_state_handle: WfcRngStateHandle,
+    max_observations: u32,
+    spent_observations: *mut u32,
 ) -> WfcObserveResult {
     let world = unsafe {
         assert!(!wfc_world_state_handle.0.is_null());
         &mut *wfc_world_state_handle.0
     };
+
+    if max_observations == 0 {
+        return match world.world_status() {
+            WorldStatus::Contradiction => WfcObserveResult::Contradiction,
+            WorldStatus::Deterministic => WfcObserveResult::Deterministic,
+            WorldStatus::Nondeterministic => WfcObserveResult::Nondeterministic,
+        };
+    }
```

Comment:
Nit, but move this after all parameters are validated, so it doesn't hide bugs in caller code.

Reply:
I attempted the earlies possible exit, but you are right.